src/lib/functions/registry.js

```diff
@@ -128,8 +117,29 @@ class FunctionsRegistry {
         return runtime.onDirectoryScan({ directory })
       }),
     )
+  }
+
+  registerFunction(name, func) {
+    if (func.isBackground && !this.capabilities.backgroundFunctions) {
+      this.logger.warn(getLogMessage('functions.backgroundNotSupported'))
+    }
+
+    this.functions.set(name, func)
+    this.buildFunctionAndWatchFiles(func)
+
+    this.logger.log(`${NETLIFYDEVLOG} ${chalk.green('Loaded')} function ${chalk.yellow(name)}.`)
+  }
+
+  async scan(directory) {
+    const directories = (Array.isArray(directory) ? directory : [directory]).filter(Boolean)
```

Comment:
[dust] Could we always pass an array to simplify the code?

Reply:
Good point! Done in https://github.com/netlify/cli/pull/2954/commits/b9a89c6fc70a43771d1553e344ee9bf009f804b4.